gwpy/timeseries/tests/test_io_gwf_lalframe.py

```diff
@@ -162,11 +167,21 @@ def test_write(tmp_path):
     assert_dict_equal(data, data2, assert_quantity_sub_equal)
 
 
-def test_write_no_ifo(tmp_path):
+@pytest.mark.parametrize(
+    'data',
+    [
+        pytest.param(
+            [],
+            marks=pytest.mark.xfail(
+                raises=RuntimeError,
+                reason='Cannot add an empty series to a frame',
+            )
+        ),
+        [1, 2, 3, 4, 5],
+    ]
+)
+def test_write_no_ifo(tmp_path, data):
     # create timeseries with no IFO
-    data = TimeSeries([1, 2, 3, 4, 5])
-    tmp = tmp_path / "test.gwf"
-    gwpy_lalframe.write(
-        {None: data},
-        tmp
-    )
+    data = TimeSeries(data, dtype=float)
+    with NamedTemporaryFile(dir=tmp_path) as f:
+        gwpy_lalframe.write({None: data}, f.name)
```

Comment:
I don't understand the switch from using `tmp_path` as it is to using `NamedTemporaryFile`, can you explain this please?

Reply:
That's me not liking hard-coded files. In this case it is harmless. I will revert the change.